### PR TITLE
add dialyzer check

### PR DESCRIPTION
**Pull request by redink**

### CODE REVIEW

**Comment by redink** [comment]
occur error when dialyzer check for gululog_r_cur module, other module all ok.

```
$ dialyzer --src -I ./include/ -c ./src/gululog_r_cur.erl 
  Checking whether the PLT /Users/redink/.dialyzer_plt is up-to-date... yes
  Proceeding with analysis...
=ERROR REPORT==== 11-May-2015::20:52:51 ===
Error in process <0.53.0> with exit value: {function_clause,[{erl_types,force_union,[unit],[{file,"erl_types.erl"},{line,2583}]},{erl_types,decorate,3,[{file,"erl_types.erl"},{line,648}]},{erl_types,decorate,3,[{file,"erl_types.erl"},{line,634}]},{dict,map_bucket,2,[{file,"d... 


dialyzer: Analysis failed with error:
{function_clause,[{erl_types,force_union,
                             [unit],
                             [{file,"erl_types.erl"},{line,2583}]},
                  {erl_types,decorate,3,[{file,"erl_types.erl"},{line,648}]},
                  {erl_types,decorate,3,[{file,"erl_types.erl"},{line,634}]},
                  {dict,map_bucket,2,[{file,"dict.erl"},{line,453}]},
                  {dict,map_bkt_list,2,[{file,"dict.erl"},{line,449}]},
                  {dict,map_bkt_list,2,[{file,"dict.erl"},{line,449}]},
                  {dict,map_seg_list,2,[{file,"dict.erl"},{line,444}]},
                  {dict,map_dict,2,[{file,"dict.erl"},{line,439}]}]}
Last messages in the log cache:
  Reading files and computing callgraph... done in 0.09 secs
  Removing edges... done in 0.32 secs
```

File: src/gululog_idx.erl [hunk 1]
```diff
@@ -36,7 +36,7 @@
 -include("gululog_priv.hrl").
 -include_lib("stdlib/include/ms_transform.hrl").
 
--opaque cache() :: ets:tid().
+-opaque cache() :: ets:tid() | atom().
```
Comment (zmshi) [inline src/gululog_idx.erl]: why?

File: src/gululog_idx.erl [hunk 2]
```diff
@@ -369,7 +369,7 @@ init_ets_from_index_file(_Version = 1, Tid, SegId, Fd) ->
 wildcard_reversed(Dir) -> gululog_name:wildcard_idx_name_reversed(Dir).
 
 %% @private Open 'raw' mode fd for writer to 'append'.
--spec open_writer_fd(IsNew :: boolean(), filename()) -> file:fd() | no_return().
+-spec open_writer_fd(IsNew :: boolean(), filename()) -> {logvsn(), file:fd()} | no_return().
```
Comment (zmshi) [inline src/gululog_idx.erl]: remove no_return()

File: src/gululog_repair.erl [hunk 3]
```diff
@@ -74,7 +74,7 @@ repair_dir(IdxFiles, SegFiles, BackupDir) ->
 %% an truncation is done for the index file as well.
 %% @end
 -spec repair_seg([filename()], [filename()], dirname(), ?undef | dirname()) ->
-        [{file_op(), filename()}].
+        no_return().
```
Comment (zmshi) [inline src/gululog_repair.erl]: what ? shouldn't it be [file_op()] ?

File: src/gululog_topic.erl [hunk 4]
```diff
@@ -129,7 +129,7 @@ truncate(#topic{dir = Dir, idx = Idx, cur = Cur} = Topic, LogId, BackupDir) ->
   end.
 
 %% @equiv delete_oldest_seg/2
--spec delete_oldest_seg(topic()) -> {topic(), [file_op()]}.
+-spec delete_oldest_seg(topic()) -> no_return().
```
Comment (zmshi) [inline src/gululog_topic.erl]: wrong!

**Comment by zmshi** [comment]
fixed issue #20 
great work!

**Pull request merged**